metrics-influxdb/src/main/java/io/dropwizard/metrics5/influxdb/GarbageFreeScheduledReporter.java

```diff
@@ -0,0 +1,199 @@
+        init(registry, filter);
+    }
+
+    protected GarbageFreeScheduledReporter(MetricRegistry registry, 
+            String name, 
+            MetricFilter filter, 
+            TimeUnit rateUnit, 
+            TimeUnit durationUnit, 
+            ScheduledExecutorService executor) {
+        super(registry, name, filter, rateUnit, durationUnit, executor);
+        init(registry, filter);
+    }
+
+    protected GarbageFreeScheduledReporter(MetricRegistry registry, 
+            String name, 
+            MetricFilter filter, 
+            TimeUnit rateUnit, 
+            TimeUnit durationUnit, 
+            ScheduledExecutorService executor, 
+            boolean shutdownExecutorOnStop) {
+        super(registry, name, filter, rateUnit, durationUnit, executor, shutdownExecutorOnStop);
+        init(registry, filter);
+    }
+
+    protected GarbageFreeScheduledReporter(MetricRegistry registry, 
+            String name,
+            MetricFilter filter, 
+            TimeUnit rateUnit, 
+            TimeUnit durationUnit, 
+            ScheduledExecutorService executor, 
+            boolean shutdownExecutorOnStop, 
+            Set<MetricAttribute> disabledMetricAttributes) {
+        super(registry, name, filter, rateUnit, durationUnit, executor, shutdownExecutorOnStop, disabledMetricAttributes);
+        init(registry, filter);
+    }
+    
+    private void init(MetricRegistry registry, MetricFilter filter) {
+        this.registry = registry;
+        this.mirror = new RegistryMirror(filter);
+        mirror.register(registry);
+    }
+
+    @Override
+    public void stop() {
+        try {
+            super.stop();
+        } finally {
+            mirror.unregister(registry);
+        }
+    }
+
+    @Override
+    @SuppressWarnings("rawtypes")
+    public void report() {
+        synchronized (this) {
+            report(mirror.gauges(), 
+                    mirror.counters(), 
+                    mirror.histograms(), 
+                    mirror.meters(), 
+                    mirror.timers());
+        }
+    }
+    
+    private static class RegistryMirror implements MetricRegistryListener {
+
+        private final MetricFilter filter;
+        @SuppressWarnings("rawtypes") // because of signature in ScheduledReporter#report(..)
+        private final ConcurrentSkipListMap<MetricName, Gauge> gauges = new ConcurrentSkipListMap<>();
+        private final ConcurrentSkipListMap<MetricName, Counter> counters = new ConcurrentSkipListMap<>();
+        private final ConcurrentSkipListMap<MetricName, Histogram> histograms = new ConcurrentSkipListMap<>();
+        private final ConcurrentSkipListMap<MetricName, Meter> meters = new ConcurrentSkipListMap<>();
+        private final ConcurrentSkipListMap<MetricName, Timer> timers = new ConcurrentSkipListMap<>();
+
+        public RegistryMirror(MetricFilter filter) {
+            this.filter = filter;
+        }
+        
+        public void register(MetricRegistry registry) {
```

Comment:
I don't see much advantage to implement `register` and `unregister` methods. ` registry.removeListener(mirror)` and `mirror.unregister(registry)` looks semantically same to me.